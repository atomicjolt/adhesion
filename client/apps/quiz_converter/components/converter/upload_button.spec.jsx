import React            from 'react';
import TestUtils        from 'react-addons-test-utils';
import UploadButton     from './upload_button';
import Stub             from '../../../../specs_support/stub';

jest.mock('../../libs/assets');

describe('Upload Button', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      isConverting: false,
      canSubmit: false,
    };

    result = TestUtils.renderIntoDocument(<Stub><UploadButton {...props} /></Stub>);
  });

  it('renders the button', () => {
    const button = TestUtils.findRenderedDOMComponentWithTag(result, 'button');
    expect(button.textContent).toContain('Upload Files');
  });

  it('shows the spinner when isConverting is true', () => {
    props.isConverting = true;
    result = TestUtils.renderIntoDocument(<Stub><UploadButton {...props} /></Stub>);
    const spinner = TestUtils.findRenderedDOMComponentWithTag(result, 'img');
    expect(spinner).toBeDefined();
  });
});
