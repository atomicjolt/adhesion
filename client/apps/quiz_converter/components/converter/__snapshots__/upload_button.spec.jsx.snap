// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Button matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadButton
    canSubmit={false}
    isConverting={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Upload Files",
      "className": "c-submit",
      "type": "submit",
    },
    "ref": null,
    "rendered": "Upload Files",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Upload Files",
        "className": "c-submit",
        "type": "submit",
      },
      "ref": null,
      "rendered": "Upload Files",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
